cactusbot/services/beam/liveloading.py

```diff
@@ -34,65 +34,32 @@ def __init__(self, channel, user):
 
         packet = await self.parse(await self.receive())
 
-        asyncio.ensure_future(self.ping(packet["data"]["pingInterval"]/1000))
-
         await super().read(handle)
 
     async def initialize(self, *interfaces):
         """Subscribe to liveloading interfaces."""
 
         if not interfaces:
-            # TODO: dict of interfaces with callbacks
-            interfaces = (
-                "channel:{channel}:update",
-                "channel:{channel}:status",
-                "channel:{channel}:followed",
-                "channel:{channel}:subscribed",
-                "user:{user}:followed",
-                "user:{user}:subscribed",
-                "user:{user}:achievement"
-            )
-
-        interfaces = tuple(
-            interface.format(channel=self.channel, user=self.user)
-            for interface in interfaces
-        )
-
-        packet = [
-            "put",
-            {
-                "method": "put",
-                "headers": {},
-                "data": {
-                    "slug": interfaces
-                },
-                "url": "/api/v1/live"
-            }
-        ]
-        self.websocket.send_str('420' + json.dumps(packet))
-
-        self.logger.info("Successfully subscribed to liveloading interfaces.")
-
-    async def ping(self, interval):
-        """Periodically ping the liveloading server to sustain connection."""
-
-        while True:
-            self.websocket.send_str('2')
-            await asyncio.sleep(interval)
-
-    async def parse(self, packet):
-        """Parse a packet from liveloading."""
-
-        match = re.match(self.RESPONSE_EXPR, packet)
+            interfaces = [
+                "channel:{channel}:update".format(channel=self.channel),
+                "channel:{channel}:status".format(channel=self.channel),
+                "channel:{channel}:followed".format(channel=self.channel),
+                "channel:{channel}:subscribed".format(channel=self.channel),
+                "user:{user}:followed".format(user=self.user),
+                "user:{user}:subscribed".format(user=self.user),
+                "user:{user}:achievement".format(user=self.user)
+            ]
+
+        packet = {
+            "type": "method",
+            "method": "livesubscribe",
+            "params": {
+                "events": interfaces
+            },
+            "id": 1
+        }
 
-        data = match.group(2)
-        if data is not None:
-            try:
-                data = json.loads(data)
-            except ValueError:
-                self.logger.exception("Invalid JSON: %s.", data)
+        self.websocket.send_str(json.dumps(packet))
 
-        return {
-            "code": match.group(1),
-            "data": data
-        }
+        print(await self.websocket.receive())
```

Comment:
Would prefer not `print`ing - if you want to keep it, `self.logger.debug`, please. :smile: